### ISSUE TITLE
Documentation Error on categorical_crossentropy

### ISSUE THREAD

**Issue by RaffEdwardBAH** [fchollet/keras#2878]
I'm working on a dataset where we want to produce a prediction for every datapoint in a sequence. For example, sequences may be 128 steps long, and we want to predict one of 7 labels for each step. So the final output size is `(nb_samples, 128, 7)`. Then I applied softmax loss (which it was not clear from documentation this would work, but the code seems to support it). 

The documentation [here](http://keras.io/objectives/#usage-of-objectives) says that the `categorical_crossentropy` loss is only for shapes `(nb_samples, nb_classes)`. However it appears to be working for my data. If it is supposed to work, the docs need to be updated / improved. If not, it should have error-ed and I don't see any available option for this scenario.
